api/v1/remotedatanodes_types.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// RemoteDataNodesSpec defines the desired state of RemoteDataNodes
+type RemoteDataNodesSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	RemoteClusterSpec *corev1.LocalObjectReference `json:"remoteClusterSpec"`
+	CommonSpec        `json:",inline"`
```

Comment:
There is mistake in remote-exec-nodes design - "CommonSpec" must be part of RemoteYtsaurusSpec, not Remote*NodesSpec.

Remote*NodesSpec should define only node instance group
and RemoteYtsaurusSpec define all common parts of cluster

Otherwise we are going to repeat everything in each spec.

Reply:
But `RemoteYtsaurusSpec` is only a description of connection endpoints to the remote cluster, and CommonSpec have instance-related stuff: ipv6/ipv4/hostNetwork/etc — all this stuff do not describe remote cluster's configuration.

Reply:
This is exactly common cluster connectivity options.

Reply:
as far as I remember
- ipv6/4 are the options for remote cluster's component DNS resolving only 
- hostNetwork — is a flag for your remote data/exec node to have hostNetwork 
Maybe there is some assumptions about remote cluster based on this parameters, but they are applied to the configured instance, not information about remote one.
Also there are ForceTCP, UseShortNames, ExtraPodAnnotations, ConfigOverrides and more which are used for the  nodes we are creating and nonrelevant to the remote ytsaurus.

Reply:
Anyways if you still think that it is a mistake let's discuss it in a separate issue, we will either fix it for both remote data exec nodes or leave as it is for both.